packages/rollup-plugin/src/rollupUtils.ts

```diff
@@ -46,10 +46,13 @@ function formatSourcemapData(
   // the same name as the js file... however we can pull the file name
   // from the sourcemap source just in case.
   let jsFilename: string
-  if (typeof sourcemap.source === 'string') {
+  if (sourcemap.source && typeof sourcemap.source === 'string') {
     const { file } = JSON.parse(sourcemap.source)
-    // The file in the source won't have the subfolder, need to add it
-    jsFilename = path.join(path.dirname(sourcemapFilename), file)
+    // The file in the source usually won't have the subfolder, we may need to add it
+    const subfolder = path.dirname(sourcemapFilename)
+    jsFilename = path.dirname(file) === subfolder 
```

Comment:
This is what I came up with:
```
function mergePaths(sourcemapFilename, source) {
  sourcemapFilename = sourcemapFilename.replace('.map', '')
  let folder1 = path.dirname(sourcemapFilename)
  const folder2 = path.dirname(source)
  if (folder1.indexOf(folder2) > -1) {
    folder1 = folder1.replace(folder2, '')
  }
  return path.join(folder1, folder2, path.basename(source))
}

// this prints: 'parent/bar/bar.js'
mergePaths('parent/bar/bar.js.map', 'bar/bar.js')
```

Reply:
> Oof, well if the file field in the sourcemap source contains a path but that path does not match the sourcemap's path, I think we've got bigger issues.

I understand, though the example I mentioned above is when the sourcemap source path is a subset of the sourcemap's path (I thought this is what @joekrump experienced).

> I've been re-reading the documentation from rollup, as well as trying to find more documentation on what the "file" field in the sourcemap should contain. Finding some interesting things.

This must not have been an interesting read 😬 

> Rollup states that The map.file property is the basename of sourcemapFile. This note is [buried under a config option](https://rollupjs.org/configuration-options/#output-sourcemapfile). "Basename" should mean the stuff after the last slash in a path -- ie just the file name with no folders. This documentation does not match the behavior @joekrump saw in https://github.com/honeybadger-io/honeybadger-js/issues/1062.

Good enough reason to write-up an issue in the rollup repo, right?

> if output.dir was set to dist, adding sourcemapFile didn't seem to do anything... the files went into the dist/ folder and were called index.js and index.js.map. The file field in the map contained index.js.

I guess this means that `output.dir` takes precedence over `output.sourcemapFile`.

> if output.dir was not set and instead output.file was set to something like indexTest.js, then the actual JS file was called indexTest.js and the sourcemap was called indexTest.js.map, but the file field within the sourcemap was what.js. This leads to sourcemaps not being uploaded, since what.js is not a real file.

So, rollup generated a bundle `indexTest.js` with source map `indexTest.js.map` from a non-existent `what.js`. Shouldn't it throw an error that `what.js` is not found?

> Would love to hear your thoughts -- I'm not actually sure 2. is much better than 1. here but I could be convinced!

At this point, I would be fine if we went forward with the simplistic approach of solution 1. However, we should acknowledge that it will break when the source maps are not in the same folder, which is possible.

Reply:
> I understand, though the example I mentioned above is when the sourcemap source path is a subset of the sourcemap's path (I thought this is what @joekrump experienced).
> 
In his example, `path.dirname(sourcemapFilename)` (aka `subfolder`) was `assets`, and the string in the `file` field was `assets/application-6016d30a.js`. So the subfolder and the path from the file field were the same. 

> > Rollup states that The map.file property is the basename of sourcemapFile. This note is [buried under a config option](https://rollupjs.org/configuration-options/#output-sourcemapfile). "Basename" should mean the stuff after the last slash in a path -- ie just the file name with no folders. This documentation does not match the behavior @joekrump saw in #1062.
> 
> Good enough reason to write-up an issue in the rollup repo, right?
> 
Agreed. So far I haven't been able to reproduce the behavior, so I can't write up an issue to rollup. Perhaps @joekrump can? 

> > if output.dir was set to dist, adding sourcemapFile didn't seem to do anything... the files went into the dist/ folder and were called index.js and index.js.map. The file field in the map contained index.js.
> 
> I guess this means that `output.dir` takes precedence over `output.sourcemapFile`.
> 
Yes, agreed. 

> > if output.dir was not set and instead output.file was set to something like indexTest.js, then the actual JS file was called indexTest.js and the sourcemap was called indexTest.js.map, but the file field within the sourcemap was what.js. This leads to sourcemaps not being uploaded, since what.js is not a real file.
> 
> So, rollup generated a bundle `indexTest.js` with source map `indexTest.js.map` from a non-existent `what.js`. Shouldn't it throw an error that `what.js` is not found?
> 
In this scenario, rollup created the bundle without complaint. Our plugin did error, since it was looking for `what.js` from the `file` field in the sourcemap and that file did not exist. 

> > Would love to hear your thoughts -- I'm not actually sure 2. is much better than 1. here but I could be convinced!
> 
> At this point, I would be fine if we went forward with the simplistic approach of solution 1. However, we should acknowledge that it will break when the source maps are not in the same folder, which is possible.
Ok, I'll go in this direction for now!

Reply:
Ok @subzero10, I simplified the logic as discussed. I went back and forth on whether to add a note to the README about this, but didn't end up adding one because we haven't yet found any scenario where the assumption that the files are in the same folder has been wrong... I don't want to add more stuff to read if it actually won't end up mattering to anyone. 

Updated the PR description rather than make a new PR since I think having this discussion thread in the PR is useful.